pint_server/app.py

```diff
@@ -117,39 +131,51 @@ def json_to_xml(json_obj, collection_name, element_name):
 
 def get_formatted_dict(obj, extra_attrs=None, exclude_attrs=None):
     obj_dict = {}
-    for attr in obj.__dict__.keys():
+    for attr, value in obj.__dict__.items():
         # FIXME(gyee): the orignal Pint server does not return the "changeinfo"
         # or "urn" attribute if it's empty. So we'll need to do the same here.
         # IMHO, I consider that a bug in the original Pint server as we should
         # return all attributes regardless whether its empty or not.
         if attr.lower() in ['urn', 'changeinfo'] and not obj.__dict__[attr]:
             continue
 
-        # NOTE: the "shape" attribute will be processed together with "type"
-        # as it is internal only
+        # NOTE(fmccarthy): shape is an internal field which, combined with
+        # the internal server type, allows us to synthesize the type value
+        # to be returned.
         if attr.lower() == 'shape':
+            # update servers have a name, region servers don't
+            if obj.__dict__.get('name'):
+                server_type = ServerType.update
+            else:
+                server_type = ServerType.region
+
+            # construct a list of elements that will make up the type value
+            type_info = [REGIONSERVER_SMT_REVERSED_MAP[server_type.value]]
+            if value:
+                type_info.append(value)
+
+            # join the elements that make up the type value with '-' to
+            # set the returned type value
+            obj_dict['type'] = '-'.join(type_info)
+
+            # TODO(fmccarthy): Do we need this???
```

Comment:
Yes. The return for a region server looks like this:

```
<server ip="54.253.118.149" name="" region="ap-southeast-2" type="regionserver-sles"/>
```
So we either have the column in the DB and empty or we add the code to the server itself. The latter would put more special conditions in the server code, which we probably want to avoid.